qep-335-adopt-wasm32-emscripten-target.md

```diff
@@ -0,0 +1,76 @@
+# QGIS Enhancement: Adopt `wasm32-emscripten` as a build target for QGIS
+
+**Date** 2025/03/10
+
+**Author** Michael Schmuki (@boardend )
+
+**Contact** michael at opengis dot ch
+
+**Version** QGIS 3.44
+
+# Summary
+
+[qgis-js](https://github.com/qgis/qgis-js) has demonstrated that it is possible to cross-compile QGIS core and it's dependencies with [Emscripten](https://emscripten.org/) to [WebAssembly](https://webassembly.org/) in order to use it on the web platform as well as other JavaScript runtimes like Node and Deno.
+
+To achieve this, [some patching](https://github.com/qgis/qgis-js/tree/main/build/vcpkg-ports/qgis-qt6) of QGIS was needed:
+- Some small patches to disable certain features that are not (yet) supported
+- Tweaking of the CMake structure/scripts
+- Some heavy, horizontal patches, for example, the exclusion of the `QgsAuthManager`
+
+Especially the horizontal patches prevent qgis-js from following the QGIS roadmap and keeping it up to date to the latest QGIS version. With this proposal we want to bring those patches upstream to the QGIS project and adopt `wasm32-emscripten` as a supported build target for QGIS.
+
+The primary goal of qgis-js is to port QGIS core to WebAssembly for rendering QGIS projects on the client side, serving as a replacement for QGIS Server. It also includes optional OpenLayers integration for seamless embedding into web applications. While qgis-js's API surface is currently limited, we aim to expand it in the future.
+
+Additionally, compiling QGIS to WebAssembly opens new possibilities, such as porting QGIS processing, bringing PyQGIS to the browser, and building full-screen QGIS apps using Qt Classic or Qt Quick (e.g., porting QField). Although this is outside the scope of qgis-js, upstreaming this work will provide a foundation for developers to explore these opportunities.
```

Comment:
Just out of curiosity, why would you like to rewrite QField with javascript. What does it improve?

Reply:
@troopa81 - It wouldn't be a rewrite in Javascript wasm is running the C++ code. Javascript bindings are only necessary if you want to embed QGIS wasm into a Javascript application (such as OpenLayers or others).

What would it enable: editing data inside the web browser without having to install software. Yes - one can edit with OpenLayers as well - but the editing tools are quite limited compared to what QField or QGIS offer.

Reply:
OK, I misunderstood the "porting QField" comment as we want to rewrite QField with qgis-js. So I understand, this is more having QField inside a browser.

Reply:
It would bring very powerful geometry editing and QGIS defined forms with expression based validation and relations to the browser. Also it lets you edit any supported vector format e.g. GeoPackage. All of those are quite hard to achieve with pure JS or existing solutions like OpenLayers.

(No concrete roadmap at the moment, but something I would like to experiment with in the feature)

Reply:
@troopa81 I guess we can resolve this one?

Reply:
yes